ports/stm32/flash.c

```diff
@@ -142,6 +148,11 @@ static uint32_t get_bank(uint32_t addr) {
 
 #elif (defined(STM32L4) && defined(SYSCFG_MEMRMP_FB_MODE)) || defined(STM32H7)
 
+// In the HAL, FLASH_BANK_SIZE reads memory address 0x08FFF80C to get the total flash size,
+// which resets the MCU.  So provide a hard-coded constant for the flash bank size.
+#undef FLASH_BANK_SIZE
+#define FLASH_BANK_SIZE (2 * 1024 * 1024 / 2)
```

Comment:
This needs to be done properly using the MPU, same as for H5, see 61339aa5062577ca1c75cb2443c78e5b5965c30b